v2/tests/log_sink_etw_int/log_sink_etw_int.c

```diff
@@ -0,0 +1,683 @@
+                current_user_data_field += sizeof(uint32_t);
+                break;
+            case TlgInINT32:
+                parsed_events[current_parsed_event_count].properties[property_index].int32_t_value = *(int32_t*)current_user_data_field;
+                current_user_data_field += sizeof(int32_t);
+                break;
+            case TlgInUINT64:
+                parsed_events[current_parsed_event_count].properties[property_index].uint64_t_value = *(uint64_t*)current_user_data_field;
+                current_user_data_field += sizeof(uint64_t);
+                break;
+            case TlgInINT64:
+                parsed_events[current_parsed_event_count].properties[property_index].int64_t_value = *(int64_t*)current_user_data_field;
+                current_user_data_field += sizeof(int64_t);
+                break;
+            }
+
+            property_index++;
+        }
+
+        parsed_events[current_parsed_event_count].property_count = property_index;
+
+        (void)InterlockedIncrement(&parsed_event_count);
+        WakeByAddressAll((void*)&parsed_event_count);
+    }
+}
+
+static char test_path[MAX_PATH];
+
+static void generate_trace_session(void)
+{
+    UUID uuid;
+    POOR_MANS_ASSERT(UuidCreate(&uuid) == RPC_S_OK);
+    int snprintf_result = snprintf(trace_session_name, sizeof(trace_session_name), "%s-%08X-%04X-%04X-%02X%02X-%02X%02X%02X%02X%02X%02X",
+        __FUNCTION__, uuid.Data1, uuid.Data2, uuid.Data3, uuid.Data4[0], uuid.Data4[1], uuid.Data4[2], uuid.Data4[3], uuid.Data4[4], uuid.Data4[5], uuid.Data4[6], uuid.Data4[7]);
+    POOR_MANS_ASSERT((snprintf_result > 0) && (snprintf_result < sizeof(trace_session_name)));
+}
+
+static TRACEHANDLE start_trace(uint8_t trace_level)
+{
+    // setup the etw consumer
+    TRACEHANDLE trace_session_handle;
+
+    (void)printf("Starting trace session %s\r\n", trace_session_name);
+
+    EVENT_TRACE_PROPERTY_DATA start_event_trace_property_data = { 0 };
+
+    start_event_trace_property_data.props.Wnode.BufferSize = sizeof(start_event_trace_property_data);
+    start_event_trace_property_data.props.Wnode.Flags = WNODE_FLAG_TRACED_GUID;
+    start_event_trace_property_data.props.LogFileMode = EVENT_TRACE_REAL_TIME_MODE;
+    start_event_trace_property_data.props.MinimumBuffers = 4;
+    start_event_trace_property_data.props.MaximumBuffers = 4;
+    start_event_trace_property_data.props.BufferSize = 64 * 1024;
+    start_event_trace_property_data.props.FlushTimer = 1;
+    start_event_trace_property_data.props.LogFileNameOffset = 0;
+    start_event_trace_property_data.props.LoggerNameOffset = offsetof(EVENT_TRACE_PROPERTY_DATA, logger_name);
+
+    POOR_MANS_ASSERT(StartTraceA(&trace_session_handle, trace_session_name, &start_event_trace_property_data.props) == ERROR_SUCCESS);
+
+    (void)strcpy(logger_name, start_event_trace_property_data.logger_name);
```

Comment:
>strcpy [](http://example.com/codeflow?start=10&length=6)

Should check that `start_event_trace_property_data.logger_name` fits in 128 bytes declared above.

Reply:
Added ASSERT